File: spring-boot-demo/src/main/scala/com/github/jccode/springbootdemo/Application.scala
package com.github.jccode.springbootdemo

import org.springframework.boot.{CommandLineRunner, SpringApplication}
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.context.ApplicationContext
import org.springframework.context.annotation.Bean
import Implicits._
import org.slf4j.LoggerFactory

@SpringBootApplication
class Application {
}

object Application {

  def main(args: Array[String]): Unit = {
    SpringApplication.run(classOf[Application], args: _*)
//    SpringApplication run classOf[Application]
    System.in.read()
  }

  @Bean
  def commandLineRunner(ctx: ApplicationContext): CommandLineRunner =
    new CommandLineRunner {
      override def run(args: String*): Unit = {
        println("Let's inspect the beans provided by Spring Boot:")
        ctx.getBeanDefinitionNames.sorted.foreach(println)
      }
    }

  @Bean
  def commandLineRunner2(ctx: ApplicationContext): CommandLineRunner = (_: Seq[String]) => {
    println("You need to import the Implicits to convert a lambda function to CommandLineRunner, if you'd like to type less words.")
  }

}

File: spring-boot-es-demo/src/main/java/com/github/jccode/esdemo/Application.java
package com.github.jccode.esdemo;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.CommandLineRunner;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.ApplicationContext;
import org.springframework.context.annotation.Bean;

/**
 * SampleApp
 *
 * @author 01372461
 */
@SpringBootApplication
public class Application {

    public static void main(String[] args) {
        SpringApplication.run(Application.class);
    }

    @Autowired
    private HighLevelApiDemo highLevelApiDemo;

    @Autowired
    private LowLevelApiDemo lowLevelApiDemo;

    @Bean
    public CommandLineRunner commandLineRunner(ApplicationContext ctx) {
        return args -> {
            System.out.println("Hi baby !");
            highLevelApiDemo.demo();
            System.out.println("--------------------");
            lowLevelApiDemo.demo();
        };
    }
}
